### PR TITLE
Tag 41

### PR DESCRIPTION
**Pull request by kacperantoniak**

### CODE REVIEW

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/TagsService.cs [hunk 1]
```diff
@@ -28,5 +29,10 @@ public async Task InstertDataAsync(Profile profile)
             await _tagsterDb.Profiles.AddAsync(profile);
             await _tagsterDb.SaveChangesAsync();
         }
+
+        public async Task<ProfileDto> GetHref(ProfileDto profileDto)
+        {
+            return profileDto;
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Infrastructure/Services/TagsService.cs:35]: there is no logic here, you just return what you pass as a parameter

File: TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            _tagService = tagService;
+        }
+        private readonly TagsterDbContext _tagsterDb;
+        public GetProfileWithTagsHandler(TagsterDbContext tagsterDb)
+            => _tagsterDb = tagsterDb;
+
+        public async Task<ProfileDto> Handle(GetProfileWithTags request, CancellationToken cancellationToken)
+        {
+            DataAccess.Entities.Profile profileInfo = _tagsterDb
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs:25]: in my opinion this logic should go to tagService.GetHref method

File: TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                .Profiles
+                .Where(profile => profile.Href.Equals(request.Href))
+                .Include(profile => profile.ProfileTags)
+                .Include(profile => profile.Name)
+                .Include(profile => profile.LastName)
+                .FirstOrDefault();
+
+            ProfileDto profileDto = new();
+            profileDto.Href = profileInfo.Href;
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs:34]: you used FirstOrDefault above so you can get NullPointerException

File: TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        public GetProfileWithTagsHandler(TagsterDbContext tagsterDb)
+            => _tagsterDb = tagsterDb;
+
+        public async Task<ProfileDto> Handle(GetProfileWithTags request, CancellationToken cancellationToken)
+        {
+            DataAccess.Entities.Profile profileInfo = _tagsterDb
+                .Profiles
+                .Where(profile => profile.Href.Equals(request.Href))
+                .Include(profile => profile.ProfileTags)
+                .Include(profile => profile.Name)
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs:29]: what is the purpose of using Include on Name and LastName?

File: TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                .Include(profile => profile.LastName)
+                .FirstOrDefault();
+
+            ProfileDto profileDto = new();
+            profileDto.Href = profileInfo.Href;
+            profileDto.Id = profileInfo.Id;
+            profileDto.LastName= profileInfo.LastName;
+            profileDto.Name = profileInfo.Name;
+            profileDto.ProfileTags = profileInfo.ProfileTags;
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs:38]: I think that you can use Select LINQ method to map Profile entity into ProfileDto, this approach can help with NullReferenceException as well

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/TagsService.cs [hunk 6]
```diff
@@ -28,5 +29,10 @@ public async Task InstertDataAsync(Profile profile)
             await _tagsterDb.Profiles.AddAsync(profile);
             await _tagsterDb.SaveChangesAsync();
         }
+
+        public async Task<ProfileDto> GetHref(ProfileDto profileDto)
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Infrastructure/Services/TagsService.cs:33]: maybe rename to GetByHref? Current name indicates that you will get href which is not true

File: TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs (hunk 4 above)
Comment (kacperantoniak) [reply TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs:29]: We want to get all info we own about the user based on our identificator (href), I think that front guys will find a way to use it to their advantage, so I'll just leave it there. If You have more concerns about this code please wait untill next meeting and I'll be happy to discuss it with You ^^

File: TagsterWebAPI/src/Tagster.Application/Dtos/ProfileDto.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Tagster.Application.Dtos
+{
+    public record ProfileDto
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string LastName { get; set; }
+        public string Href { get; set; }
+        public ICollection<Tag> ProfileTags { get; set; }
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Application/Dtos/ProfileDto.cs:12]: in my opinion ProfileTags should be DTO type as well

File: TagsterWebAPI/src/Tagster.Application/Dtos/ProfileDto.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Tagster.Application.Dtos
+{
+    public record ProfileDto
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string LastName { get; set; }
+        public string Href { get; set; }
+        public ICollection<TagDto> ProfileTags { get; set; }
```
Comment (flapek) [inline TagsterWebAPI/src/Tagster.Application/Dtos/ProfileDto.cs:12]: Rename 'ProfileTags' to 'Tags'

**Review by flapek — CHANGES_REQUESTED** [review]

**Review by flapek — APPROVED** [review]

**Pull request merged**